src/services/copyConfig/policies/policyOptions.js

```diff
@@ -31,158 +31,188 @@ class PolicyOptions extends Options {
-          value: true,
-          tooltip: 'POLICIES_DOUBLE_OPT_IN',
-        },
-        {
-          id: `p${PolicyOptions.#preferencesCenter}`,
-          name: PolicyOptions.#preferencesCenter,
-          value: true,
-          tooltip: 'POLICIES_PREFERENCES_CENTER',
-        },
-      ],
+      branches: [],
     })
     this.#policy = policy
   }
 
   getConfiguration() {
     return this.#policy
   }
-  removeAccountOptions(info) {
-    return this.removeInfo(PolicyOptions.#accountOptions, info)
-  }
-  removeAuthentication(info) {
-    return this.removeInfo(PolicyOptions.#authentication, info)
-  }
-  removeDoubleOptIn(info) {
-    return this.removeInfo(PolicyOptions.#doubleOptIn, info)
-  }
-  removePreferencesCenter(info) {
-    return this.removeInfo(PolicyOptions.#preferencesCenter, info)
-  }
-  removeCodeVerification(info) {
-    return this.removeInfo(PolicyOptions.#codeVerification, info)
-  }
 
-  removeEmailNotification(info) {
-    return this.removeInfo(PolicyOptions.#emailNotifications, info)
-  }
+  addSupportedPolicies(response) {
+    if (response.accountOptions) {
+      this.options.branches.push({
+        id: PolicyOptions.#accountOptions,
+        name: PolicyOptions.#accountOptions,
+        propertyName: PolicyOptions.#accountOptions,
+        value: true,
+        tooltip: 'POLICIES_ACCOUNT_OPTIONS',
+      })
+    }
+    if (response.authentication) {
+      this.options.branches.push({
+        id: PolicyOptions.#authentication,
+        name: PolicyOptions.#authentication,
+        propertyName: PolicyOptions.#authentication,
+        value: true,
+        tooltip: 'POLICIES_AUTHENTICATION',
+      })
+    }
+    if (response.codeVerification) {
+      this.options.branches.push({
+        id: `${PolicyOptions.#idPrefix}${PolicyOptions.#codeVerification}`,
+        name: PolicyOptions.#codeVerification,
+        propertyName: PolicyOptions.#codeVerification,
+        value: true,
+        tooltip: 'CODE_VERIFICATION',
+      })
+    }
+    if (response.emailNotifications) {
+      this.options.branches.push({
+        id: PolicyOptions.#emailNotifications,
+        name: PolicyOptions.#emailNotifications,
+        propertyName: PolicyOptions.#emailNotifications,
+        value: true,
 
-  removeEmailVerification(info) {
-    return this.removeInfo(PolicyOptions.#emailVerification, info)
-  }
+        tooltip: 'POLICIES_EMAIL_NOTIFICATIONS',
+      })
+    }
+    if (response.emailVerification) {
+      const evOption = {
+        id: `${PolicyOptions.#idPrefix}${PolicyOptions.#emailVerification}`,
+        name: PolicyOptions.#emailVerification,
+        propertyName: PolicyOptions.#emailVerification,
+        value: true,
+        tooltip: 'POLICIES_EMAIL_VERIFICATION',
+      }
+      if (response.emailVerification.nextURL) {
+        evOption['branches'] = [
+          {
+            id: `${evOption.id}-Link`,
+            name: 'Include Customize Redirection URL',
```

Comment:
Can we improve this in a way that it doesn't require to have this text directly here? Maybe storing it in "PolicyOptions."?

Reply:
The text was added to the translations file. We are now using translation on this node.